File: katayunos/src/test/scala/FizzBuzzTest.scala
import org.scalatest.FunSuite

class FizzBuzzTest extends FunSuite {
  val r = FizzBuzz.fizzBuzz

  test("Not divisible between 3 or 5 returns the number") {
    assert (r(0) == "1")
  }

  test("Divisible between 3 returns 'Fizz'") {
    assert (r(2) == "Fizz")
  }

  test("If number contains 3 returns 'Fizz'") {
    assert (r(30) == "Fizz")
  }

  test("Divisible between 5 returns 'Buzz'") {
    assert (r(4) == "Buzz")
  }

  test("If number contains 5 returns 'Buzz'") {
    assert (r(51) == "Buzz")
  }

  test("If number contains 5 and 3 returns 'FizzBuzz'") {
    assert (r(52) == "FizzBuzz")
    assert (r(34) == "FizzBuzz")
  }

  test("Divisible between 5 and 3 returns 'FizzBuzz'") {
    assert (r(14) == "FizzBuzz")
  }
}

File: katayunos/src/test/java/fizzbuzz/java/FizzBuzzTest.java
package fizzbuzz.java;

import static fizzbuzz.java.FizzBuzz.fizzBuzz;
import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class FizzBuzzTest {

    @Test public void notDivisibleBetween3Or5ReturnsTheNumber () {
        assertEquals (fizzBuzz ().get (0), "1");
    }

    @Test public void divisibleBetween3ReturnsFizz () {
        assertEquals (fizzBuzz ().get (2), "Fizz");
    }

    @Test public void ifNumberContains3ReturnsFizz () {
        assertEquals (fizzBuzz ().get (30), "Fizz");
    }

    @Test public void divisibleBetween5ReturnsBuzz () {
        assertEquals (fizzBuzz ().get (4), "Buzz");
    }

    @Test public void ifNumberContains5ReturnsBuzz () {
        assertEquals (fizzBuzz ().get (51), "Buzz");
    }

    @Test public void ifNumberContains5And3ReturnsFizzBuzz () {
        assertEquals (fizzBuzz ().get (52), "FizzBuzz");
        assertEquals (fizzBuzz ().get (34), "FizzBuzz");
    }

    @Test public void divisibleBetween5And3ReturnsFizzBuzz () {
        assertEquals (fizzBuzz ().get (14), "FizzBuzz");
    }
}
